### PR TITLE
Use inline array functions for returned arrays

### PR DESCRIPTION
**Pull request by coolo**
Inspired by @bgeuken's comment:
 https://github.com/openSUSE/open-build-service/pull/6425#discussion_r238106145

The performance gain is most likely unmeasurable but it's still
a good default not to waste another array allocation

### CODE REVIEW

File: src/api/app/controllers/webui/project_controller.rb [hunk 1]
```diff
@@ -779,7 +779,7 @@ def find_maintenance_infos
 
     @is_maintenance_project = @project.is_maintenance?
     if @is_maintenance_project
-      @open_maintenance_incidents = @project.maintenance_incidents.pluck('projects.name').sort.uniq
+      @open_maintenance_incidents = @project.maintenance_incidents.pluck('projects.name').sort!.uniq!
```
Comment (bgeuken) [inline src/api/app/controllers/webui/project_controller.rb]: There are some cases were the 'bang' version is returning nil when the object wasn't changed, [uniq](https://ruby-doc.org/core-2.5.0/Array.html#method-i-uniq-21) is one of them.

Comment (coolo) [reply src/api/app/controllers/webui/project_controller.rb]: yeah, mean pitfall - will have to revert some cases :)

File: src/api/app/jobs/consistency_check_job.rb [hunk 2]
```diff
@@ -234,24 +234,24 @@ def dir_to_array(xmlhash)
...
       array << e['name']
     end
-    array.sort
+    array.sort!
   end
 
   def hash_diff(a, b)
     # ignore the order inside of the hash
-    (a.keys.sort | b.keys.sort).each_with_object({}) do |diff, k|
+    (a.keys.sort! | b.keys.sort!).each_with_object({}) do |diff, k|
```
Comment (Ana06) [inline src/api/app/jobs/consistency_check_job.rb]: [nitpick] I wonder if this is what we really want here, or this: `(a.keys | b.keys).sort!`

File: src/api/app/models/project.rb [hunk 3]
```diff
@@ -1587,8 +1587,8 @@ def self.validate_repository_xml_attribute(request_data, project_name)
   end
 
   def get_removed_repositories(request_data)
-    new_repositories = request_data.elements('repository').map(&:values).flatten
-    old_repositories = repositories.all.map(&:name)
+    new_repositories = request_data.elements('repository').map!(&:values).flatten
+    old_repositories = repositories.map(&:name)
```
Comment (Ana06) [inline src/api/app/models/project.rb]: why not `!` here?

Comment (coolo) [reply src/api/app/models/project.rb]: This actually sounds like a job for pluck

Comment (coolo) [reply src/api/app/models/project.rb]: But to answer your question:
```
undefined method `map!' for #<Repository::ActiveRecord_Associations_CollectionProxy:0x00000000085d93c0>

File: src/api/app/jobs/consistency_check_job.rb (hunk 2 above)
Comment (coolo) [reply src/api/app/jobs/consistency_check_job.rb]: I agree.

**Pull request merged**